examples/livehunt_network_watch.py

```diff
@@ -0,0 +1,224 @@
+          print(f'Error updating {name}: {e}')
+
+      else:
+        ruleset = vt.Object(
+            obj_type='hunting_ruleset',
+            obj_attributes={
+                'name': name,
+                'match_object_type': task.get('kind'),
+                'enabled': True,
+                'tags': ('autogenerated',),
+                'rules': task.get('rules')})
+        try:
+          await client.post_object_async(
+              path='/intelligence/hunting_rulesets', obj=ruleset)
+          print(f'Ruleset {name} created.')
+        except vt.error.APIError as e:
+          print(f'Error saving {name}: {e}')
+
+      queue.task_done()
+
+
+def main():
+  parser = argparse.ArgumentParser(description=(
+     'Manage a set of YARA nethunting rules by adding/removing domains.'))
+  parser.add_argument('-l', '--list', action='store_true',
+                      help='List current monitored domains.')
+  parser.add_argument('-a', '--add-domain',
+                      help='Add a domain to the list.')
+  parser.add_argument('-d', '--delete-domain',
+                      help='Remove a domain from the list.')
+  parser.add_argument('--workers', type=int, required=False, default=4,
+                      help='number of concurrent workers')
+  args = parser.parse_args()
+
+
+  if os.environ.get(API_KEY_ENV_VAR) is None:
+    print(f'Please set {API_KEY_ENV_VAR} environment variable')
+    return
+
+  rulesets = get_rulesets()
+  if not rulesets and not args.add_domain:
+    print('* Empty domain list, use --add-domain domain.tld to register one')
+    sys.exit(1)
+
+  domains = rulesets.get('url', {}).get('domains', [])
+  if args.list:
+    if not domains:
+      print('* Empty domain list, use --add-domain domain.tld to register one')
+      sys.exit(0)
+
+    print('Currently monitored domains:')
+    for domain in domains:
+      print(f'- {domain}')
+    sys.exit(0)
+
+  new_domain_list = copy.copy(domains)
+  if args.add_domain:
+    new_domain_list.append(args.add_domain)
+
+  if args.delete_domain:
+    if not args.delete_domain in new_domain_list:
+      print(f'* {args.delete_domain} not in list')
+      sys.exit(1)
+    new_domain_list.remove(args.delete_domain)
+
+  new_domain_list = list(set(new_domain_list))
+  new_domain_list.sort()
+
+  if new_domain_list != domains:
+    print('Updating monitored list:')
+    for domain in new_domain_list:
+      print(f'- {domain}')
+
+    # Update the rulesets
+    loop = asyncio.get_event_loop()
+    queue = asyncio.Queue(loop=loop)
+
+    loop.create_task(build_rulesets(queue, rulesets, new_domain_list))
+
+    worker_tasks = []
+    for _ in range(args.workers):
+      worker_tasks.append(
+          loop.create_task(upload_rulesets(queue)))
+
+    # Wait until all worker tasks has completed.
+    loop.run_until_complete(asyncio.gather(*worker_tasks))
+    loop.close()
+
+  else:
+    print('Nothing to do')
+
+if __name__ == '__main__':
+  main()
```

Comment:
The recommended approach is using asyncio.run: https://docs.python.org/3/library/asyncio-runner.html#asyncio.run